### DISCUSSION TITLE
confused over run as system account/sccm

### DISCUSSION THREAD

**Discussion by brianmantler** [microsoft/winget-cli#2338] [Q&A] [answered]
I've read everything I can find and am confused over how winget can currently run as system in sccm, and how it will work in the future.

What I am ultimately looking for is a solution to run "winget upgrade --all" in the enterprise.

Is there a current solution to do this?  https://github.com/Romanitho/Winget-Install or https://github.com/Romanitho/Winget-AutoUpdate might be able to do this.  But so far my early testing hasn't worked

However will winget eventually be able to do this in SCCM?  We aren't migrated to intune yet, and truthfully I don't have much understanding of how Intune/MECM work with each other. 

Thanks

**Comment by denelon** [comment, marked as answer]
The Windows Package Manager is delivered as a part of the "App Installer". It's an MSIX package so there is no system level concept. MSIX packages are installed to a virtualized registry. The work we're doing as a part of the Intune integration will support remote access in system context for Intune as well as other MDM providers.

* #215 

We're building a NuGet package that will run "in process" so the COM API will be exposed.
